packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -98,5 +167,143 @@ export async function forceLightningLwcPreview(sourceUri: vscode.Uri) {
+  if (desktopSelected) {
+    await startServer(true, fullUrl, startTime);
+    return;
+  }
+
+  let target: string = platformSelection.defaultTargetName;
+  let placeholderText =
+    platformSelection.id === PreviewPlatformType.Android
+      ? nls.localize('force_lightning_lwc_android_target_default')
+      : nls.localize('force_lightning_lwc_ios_target_default');
+  const rememberDeviceConfigured =
+    getWorkspaceSettings().get(rememberDeviceKey) || false;
+  const lastTarget = getRememberedDevice(platformSelection);
+
+  // Remember device setting enabled and previous device retrieved.
+  if (rememberDeviceConfigured && lastTarget) {
+    const message =
+      platformSelection.id === PreviewPlatformType.Android
+        ? 'force_lightning_lwc_android_target_remembered'
+        : 'force_lightning_lwc_ios_target_remembered';
+    placeholderText = nls.localize(message, lastTarget);
+    target = lastTarget;
+  }
+  const targetName = await vscode.window.showInputBox({
+    placeHolder: placeholderText
+  });
+
+  if (targetName === undefined) {
+    vscode.window.showInformationMessage(
+      platformSelection.id === PreviewPlatformType.Android
+        ? nls.localize('force_lightning_lwc_android_device_cancelled')
+        : nls.localize('force_lightning_lwc_ios_device_cancelled')
+    );
+    return;
+  }
+  await startServer(false, fullUrl, startTime);
+
+  // New target device entered
+  if (targetName !== '') {
+    updateRememberedDevice(platformSelection, targetName);
+    target = targetName;
+  }
+
+  const mobileCancellationTokenSource = new vscode.CancellationTokenSource();
+  const mobileCancellationToken = mobileCancellationTokenSource.token;
+  const targetUsed = target || platformSelection.defaultTargetName;
+  const command = new SfdxCommandBuilder()
+    .withDescription(commandName)
+    .withArg(sfdxMobilePreviewCommand)
+    .withFlag('-p', platformSelection.platformName)
+    .withFlag('-t', targetUsed)
+    .withFlag('-n', componentName)
+    .withFlag(
+      '--loglevel',
+      getWorkspaceSettings().get(logLevelKey) || defaultLogLevel
+    )
+    .build();
+
+  const mobileExecutor = new CliCommandExecutor(command, {
+    env: { SFDX_JSON_TO_STDOUT: 'true' }
+  });
+  const execution = mobileExecutor.execute(mobileCancellationToken);
+  telemetryService.sendCommandEvent(logName, startTime);
+  channelService.streamCommandOutput(execution);
+  channelService.showChannelOutput();
+
+  execution.processExitSubject.subscribe(async exitCode => {
+    if (exitCode !== 0) {
+      const message =
+        platformSelection.id === PreviewPlatformType.Android
+          ? nls.localize('force_lightning_lwc_android_failure', targetUsed)
+          : nls.localize('force_lightning_lwc_ios_failure', targetUsed);
+      showError(new Error(message), logName, commandName);
+
+      // Error code 127 means the lwc on mobile sfdx plugin is not installed.
+      if (exitCode === 127) {
```

Comment:
It would be good to also push this message to the error message surfaced on the lower right corner similar to how we do it for the Diff plugin. You can look at that code here https://github.com/forcedotcom/salesforcedx-vscode/blob/0e506ab11331b4345d792562456be0daa8b61e25/packages/salesforcedx-vscode-core/src/commands/forceSourceDiff.ts#L76

![mobile-plugin-missing-error](https://user-images.githubusercontent.com/1041842/83435999-fcaf1580-a3f1-11ea-94c1-c552183652fd.png)

![diff-plugin-missing-error](https://user-images.githubusercontent.com/1041842/83436014-0173c980-a3f2-11ea-8613-e104b53cf9ed.png)